amahi/ios
# HDA PIN Authorisation Feature

Based on issue #221 

If you select the default hda(welcome to Amahi), the app will not ask for a pin and you can browse the files. 

Otherwise the app would ask for a PIN and if authorised, the auth token will be stored on the device so the user doesn’t need to authenticate twice. 

You can log out from an HDA on the shares screen and the previously stored token will be deleted.

Also, in case the stored token would get expired, the app will notify the user and log out.


**HOW TO TEST**

SETUP

Please contact me for the fs binary file. The fs binary file that I provide is an amahi web server which only accepts "1234" pin, any other pins would get denied. So, you'd need to enter that pin in the app in order to access the HDA's shares and files.

Before getting started: 

1. Find your local IP address and append the 4563 port and write it down. E.g.: 192.168.178.83:4563

2. Create a folder called "server" somewhere on your hard drive and inside that folder create a folder called images. This will be the share inside the HDA. 

3. Write down the path to that server folder. E.g.: Users/marton/Desktop/Server


INSTRUCTIONS


1. Open terminal

2. Run this command to give permission for the next instruction:
chmod a+x fs

3. Navigate to the folder where you placed the binary fs file

4. Specify where your server files are and your local IP address and run the following command:
./fs -d=3 -k=abc -r=/Users/marton/Desktop/server -l 192.168.178.83:4563

k= -> this can be any string, but we need to take note of it so that the app can find the HDA
-r= -> this is the path to the server folder that you created earlier

Now your HDA is up and running.

Next, open the Xcode project and navigate to ServerPresenter.swift file and go to line 48. Please uncomment the next 5 lines so the app can see the HDA you created before. The token of the server is the string you specified in the above command (e.g.: "abc")

That's it, now the app should see your hda and you can test out the authentication by providing the correct pin(1234).
